### PR TITLE
display displayname on federated shares

### PR DESCRIPTION
**Pull request by ArtificialOwl**
retrieve data from lookup-server if available when displaying shares.
includes some caching

also needed:

- lus: https://github.com/nextcloud/lookup-server/pull/100
- gss: https://github.com/nextcloud/globalsiteselector/pull/83

210675677-e2c390ea-9936-40cc-93d3-78e0bb0674d6
210675676-1be333c9-e8ba-4d7b-a13b-017369ffcfcc
210675669-4bdc2059-c86a-488c-96a0-28be48695e2d

### CODE REVIEW

File: apps/federatedfilesharing/lib/OCM/CloudFederationProviderFiles.php [hunk 1]
```diff
@@ -335,13 +341,13 @@ private function mapShareTypeToNextcloud($shareType) {
 		return $result;
 	}
 
-	private function notifyAboutNewShare($shareWith, $shareId, $ownerFederatedId, $sharedByFederatedId, $name): void {
+	private function notifyAboutNewShare($shareWith, $shareId, $ownerFederatedId, $sharedByFederatedId, $name, $displayName): void {
```
Comment (come-nc) [inline apps/federatedfilesharing/lib/OCM/CloudFederationProviderFiles.php:344]:
```suggestion
	private function notifyAboutNewShare($shareWith, $shareId, $ownerFederatedId, $sharedByFederatedId, $name, string $displayName): void {
```
And might as well type all parameters, this is a private function.

**Review by come-nc — APPROVED** [review]

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 2]
```diff
@@ -360,6 +376,102 @@ private function getDisplayNameFromAddressBook(string $query, string $property):
 		return $query;
 	}
 
+
+	/**
```
Comment (ArtificialOwl) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:380]: add some comment on how this method works ...

**Comment by blizzz** [comment]
what's with all the psalm warnings?

**Review by AndyScherzinger — APPROVED** [review]

**Comment by AndyScherzinger** [comment]
@ArtificialOwl can you take care of the manual backports? 🥇

**Pull request merged**